frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -598,6 +598,33 @@ class Worksheet extends React.Component {
             return;
         }
         const item = this.refs.list.refs['item' + index];
+
+        // Make the judgement only if the shouldScroll parameter has the default value of `True`
+        // Logic: Only scroll when the target element is completely not on the screen
+        if (shouldScroll) {
+            const element = subIndex
+                ? $(`#codalab-worksheet-item-${index}-subitem-${subIndex}`)
+                : $(`#codalab-worksheet-item-${index}`);
+
+            function isOnScreen(element) {
+                if (element.offset() === undefined) return false;
+                var elementOffsetTop = element.offset().top;
+                // For result table which should first scroll itself when a row inside it is not visible anymore.
+                // The max height for the result table is 500.
```

Comment:
Is there a way to factor out "500" as a constant, so we can easily change it later without having to go through this code?

Reply:
I guess (<= 262 ||  262 + >= MAX_TABLE_HEIGHT) would do (line 614 below)